airflow/api_fastapi/core_api/routes/public/dag_run.py

```diff
@@ -147,3 +156,70 @@ def patch_dag_run(
     dag_run = session.get(DagRun, dag_run.id)
 
     return DAGRunResponse.model_validate(dag_run, from_attributes=True)
+
+
+@dag_run_router.get("/", responses=create_openapi_http_exception_doc([401]))
+async def get_dag_runs(
+    dag_id: str,
+    limit: QueryLimit,
+    offset: QueryOffset,
+    logical_date: Annotated[
+        RangeFilter, Depends(datetime_range_filter_factory("logical_date", DagRun, "execution_date"))
+    ],
+    start_date_range: Annotated[RangeFilter, Depends(datetime_range_filter_factory("start_date", DagRun))],
+    end_date_range: Annotated[RangeFilter, Depends(datetime_range_filter_factory("end_date", DagRun))],
+    update_at_range: Annotated[RangeFilter, Depends(datetime_range_filter_factory("updated_at", DagRun))],
+    state: QueryDagRunStateFilter,
+    order_by: Annotated[
+        SortParam,
+        Depends(
+            SortParam(
+                [
+                    "id",
+                    "state",
+                    "dag_id",
+                    "execution_date",
+                    "dag_run_id",
+                    "start_date",
+                    "end_date",
+                    "updated_at",
+                    "external_trigger",
+                    "conf",
+                ],
+                DagRun,
+            ).dynamic_depends(default="id")
+        ),
+    ],
+    session: Annotated[Session, Depends(get_session)],
+    # fields: list[str] | None = Query(None),
```

Comment:
@pierrejeambrun, @bugraoz93 , looking for ideas on how to implement fields to return.

Reply:
Due to its type and purpose, this should be similar to `update_mask` (even though it is reversed regarding `masking`).  We will get in `explode=true`, similar to update_mask again `....?fields=item1&fields=item2`. 

As far as I see there isn't a way to exclude fields during `validation` and doing `dump` and `validate` won't work until we have a dynamically generated `Pydantic` model with the selected fields making the validation on return. I saw a discussion about including this capability. There are also some workarounds for people there. Maybe we can also have a generic exclusion method for `Pydantic` models until it is provided fully.
https://github.com/pydantic/pydantic/issues/9573

Reply:
For now you can just skip `fields`. Because it's an issue to implement and dynamically change the returned type. (sub fields etc...), which isn't great for the client.

This is mentioned here https://github.com/apache/airflow/issues/43378#issue-2613951437, as one of the breaking change.

We can either fix that later before 3.0 release, or accept that breaking change. In the mean time I would suggest to just remove it and don't bother with it.